test/models/activation_key_test.rb

```diff
@@ -71,5 +72,30 @@ def setup
       new_key.max_content_hosts = 100
       assert !new_key.valid?
     end
+
+    def test_search_name
+      activation_keys = ActivationKey.search_for("name = \"#{@dev_staging_view_key.name}\"")
+      assert_includes activation_keys, @dev_staging_view_key
+    end
+
+    def test_search_organization_id
+      activation_keys = ActivationKey.search_for("organization_id = \"#{@dev_staging_view_key.organization.id}\"")
+      assert_includes activation_keys, @dev_staging_view_key
+    end
+
+    def test_search_environment
+      activation_keys = ActivationKey.search_for("environment = \"#{@dev_staging_view_key.environment.name}\"")
+      assert_includes activation_keys, @dev_staging_view_key
+    end
+
+    def test_search_content_view
+      activation_keys = ActivationKey.search_for("content_view = \"#{@dev_staging_view_key.content_view.name}\"")
+      assert_includes activation_keys, @dev_staging_view_key
+    end
+
+    def test_search_content_view_id
+      activation_keys = ActivationKey.search_for("content_view_id = \"#{@dev_staging_view_key.content_view.id}\"")
+      assert_includes activation_keys, @dev_staging_view_key
+    end
   end
```

Comment:
Do you think it would be a good idea to add some checks that all the activation keys are not being returned which would still pass these tests?